Test

GerashenkoVladimir opened the pull request.

GerashenkoVladimir committed:
feature(core): Loader

GerashenkoVladimir committed:
fix(core): Loader::autoload()

GerashenkoVladimir committed:
style(index.php): fix misprints
fix(src/Blog/Controller): deleted test.php

GerashenkoVladimir committed:
style(myblog): code formatting

GerashenkoVladimir committed:
feature(router): on development stage

GerashenkoVladimir committed:
feature (router): the continued development

GerashenkoVladimir committed:
feature(router): developing

GerashenkoVladimir committed:
feature(router): developing

GerashenkoVladimir committed:
feature(router): developing

GerashenkoVladimir committed:
feature(router): developing

GerashenkoVladimir committed:
feature(core): Router.php test version

GerashenkoVladimir committed:
fix(Router.php): fixed "if" blocks

GerashenkoVladimir committed:
style(Router.php): fixed code format

GerashenkoVladimir committed:
fix(Router.php): add controller and action check

GerashenkoVladimir committed:
feature(Exceptions/FileNotFoundException): test Exceptions/FileNotFoundException

GerashenkoVladimir committed:
feature(Exceptions/RouterExceptions): exceptions in Router.php

GerashenkoVladimir committed:
fix(Exceptions/RouterExceptions): simplification of logic

GerashenkoVladimir committed:
feature(Registry/Registry): added Registry.php

GerashenkoVladimir committed:
feature(Registry/Registry): some fixes

GerashenkoVladimir committed:
feature(Request/Request) added Request.php

GerashenkoVladimir committed:
fix(Request/Request): added Request::getElement() and Request::getSessionData()

GerashenkoVladimir committed:
fix(Registry/Registry): fixed error of Registry::__construct()

GerashenkoVladimir committed:
feature(Request/Request): added Request::getHeaders()

GerashenkoVladimir committed:
docs(core): add comments

GerashenkoVladimir committed:
feature(DataBase/DataBase; Response/Response): in developing

GerashenkoVladimir committed:
feature(DataBase/DataBase): developing

GerashenkoVladimir committed:
feature(DataBase/DataBase): DataBase.php

GerashenkoVladimir committed:
feature(DataBase/DataBase): complete developing DataBase/DataBase

GerashenkoVladimir committed:
fix(DataBase/DataBase): Deleted try{}catch(){} blocks

GerashenkoVladimir committed:
feature(ActiveRecord): ActiveRecord developing

GerashenkoVladimir committed:
feature(core): developing 05.02.2016

GerashenkoVladimir committed:
feature(DI/Service): created DI/Service

GerashenkoVladimir committed:
feature(Sessions/Sessions): created Sessions/Sessions

GerashenkoVladimir committed:
pull commit

GerashenkoVladimir committed:
Merge branch 'test' of https://github.com/GerashenkoVladimir/myblog into test

vkopitsa commented on app/config/config.php:
```diff
@@ -4,14 +4,20 @@
+    'pdo'         => array(
+        'dns'      => 'mysql:dbname=education;host=127.0.0.1',
+        'user'     => 'education',
+        'password' => 'gfhjkm1987'
```
Лучше хранить в отдельном файле и файл положить под гит игнор

ddavidov replied:
+

vkopitsa replied:
:+1:

GerashenkoVladimir replied:
fixed

vkopitsa commented on framework/Application.php:
```diff
@@ -1,18 +1,35 @@
+        } catch (RouterException $e){
+            echo $e;
+        } catch (\Exception $e){
+            echo $e;
```
?

vkopitsa commented:
1. Добавить описание методов/классов
2. Подчистить код, много закомментированного

vkopitsa commented on framework/Loader.php:
```diff
@@ -1,32 +1,76 @@
-    }
 
-}
+        $filePath = __DIR__.'/../'.lcfirst($classname).'.php';
```
Можно сделать что-то на подобие https://github.com/yiisoft/yii2/blob/master/framework/BaseYii.php#L23 чтобы не писать везде **DIR**.'/../

vkopitsa commented on framework/Request/Request.php:
```diff
@@ -0,0 +1,207 @@
+    public function __construct()
+    {
+        $this->registry       = Registry::getInstance();
+        $this->allHeaders     = getallheaders();
```
Эти обработчики можно записать в get методах, чтобы лишний раз не выполнять https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L230

vkopitsa commented on framework/Router/Router.php:
```diff
@@ -0,0 +1,218 @@
+    public function generateURL($routeName)
+    {
+        if (isset($this->registry['config']['routes'][$routeName]['pattern'])) {
+            return "http://{$this->registry['request']->getHTTPHost()}{$this->registry['config']['routes'][$routeName]['pattern']}";
```
а вдруг захочешь использовать https

vkopitsa commented on src/Blog/views/layout.html.php:
```diff
@@ -66,15 +71,17 @@
-            <div class="alert alert-<?php echo $type==='error'?'danger':$type?> alert-dismissible" role="alert">
-                <button type="button" class="close" data-dismiss="alert"><span aria-hidden="true">&times;</span><span
-                        class="sr-only">Close</span></button>
-                <?php echo $msg; ?>
```
Для безопасности используй htmlspecialchars , https://github.com/yiisoft/yii2/blob/ef5ce2d4c12bbf8db3179f075e3c18a414b9e9d2/framework/helpers/BaseHtml.php#L102   https://github.com/yiisoft/yii2/blob/50cd3e1123a7461a672fc2be857ffad5d85104b4/build/controllers/views/translation/report_html.php#L32

GerashenkoVladimir committed:
fix(core): fixed Router/Router, Application and some small fixes.

ddavidov commented on framework/Application.php:
```diff
@@ -1,18 +1,67 @@
+        $registry['request']  = new Request();
+        $registry['config']   = require_once('../app/config/config.php');
+        $registry['dataBase'] = DataBase::getInstance();
+        $registry['router']   = $router = new Router();
```
Всё, что выше можно вынести в конструктор

ddavidov replied:
А если уже есть DI service, почему не применял сразу? )

GerashenkoVladimir replied:
Сначала потому что его не было)
Если я правильно понял то $registry[] использовать как контейнер который будет использовать DI service как хранилище?

GerashenkoVladimir committed:
Delete config.php

GerashenkoVladimir committed:
fixes(Application,config/config): fixed Application::__construct() and database config.

GerashenkoVladimir committed:
fix(core): fixed config/config

GerashenkoVladimir committed:
fix(Renderer/Renderer): add helpers.php (generateToken() function)

GerashenkoVladimir committed:
feature(DI/Service): Completely rebuilt

GerashenkoVladimir committed:
Удалил лишний код в Application.php

GerashenkoVladimir committed:
fix(DI/Service): fixed some errors

GerashenkoVladimir committed:
feature(core): added login, sign in and logout. And some "small" fixes/

GerashenkoVladimir committed:
faeture(core): added FlushMessenger/FlushMessenger

GerashenkoVladimir committed:
fix(FlushMessenger/FlushMessenger): fixed FlushMessenger/FlushMessenger::__constructor(),
add FlushMessenger/FlushMessenger::setMessage() and FlushMessenger/FlushMessenger::getMessages()

GerashenkoVladimir committed:
feature(core): add Request/Request::checkToken()
fix(core): many small fixes

GerashenkoVladimir committed:
feature(core):ErrorHandler/ErrorHandler

GerashenkoVladimir committed:
feature(core): Exception/MainException
fix(core):exception

GerashenkoVladimir committed:
fix(core): some fixes

GerashenkoVladimir committed:
fix(core): fixed Exception/MainException and exception handlers

GerashenkoVladimir committed:
fix(Response/Response): add some methods

GerashenkoVladimir committed:
fix(core): add Closure function in helpers.php

GerashenkoVladimir committed:
docs(core): add comments

GerashenkoVladimir committed:
docs(src/Blog): add comments;
feature(core): JsonResponse

ddavidov commented on app/config/routes.php:
```diff
@@ -2,36 +2,43 @@
 
 return array(
     'home'           => array(
+        '_name'      => 'home',
```
Я так смотрю, ключи в качестве имён заюзать не получилось?
Ну то не критично.

GerashenkoVladimir replied:
та что то как то не придумал сразу, но если надо попробую поковырять

ddavidov commented:
Смотрю, что всё вроде готово, много работы проделано. Если рабочий вариант - можно сливать в мастер уже.
